src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
+const MAX_HEALTH = 100;
+const DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  static alive = [];
+  health = MAX_HEALTH;
+
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(selectedHerbivore) {
+    if (
+      !selectedHerbivore.hidden &&
+      Object.hasOwn(selectedHerbivore, 'hidden')
```

Comment:
u can use `selectedHerbivore instanceof Herbivore` for i, cuz in the future, carnivore can learn to hide too))